jobserver/views/workspaces.py

```diff
@@ -271,28 +285,69 @@ def post(self, request, *args, **kwargs):
         return redirect(workspace)
 
 
-class WorkspaceOutputList(View):
-    def get(self, request, *args, **kwargs):
-        """
-        Orchestrate viewing of a Release in the SPA
+class WorkspaceCurrentOutputsDetail(View):
+    """
+    Orchestrate viewing of the Workspace's outputs in the SPA
 
-        We consume two URLs with one view, because we want to both do
-        permissions checks on the Release but also load the SPA for any given
-        path under a Release.
-        """
+    We consume two URLs with one view, because we want to both do permissions
+    checks on the Workspace but also load the SPA for any given path under the
+    Workspace.
+    """
+
+    def get(self, request, *args, **kwargs):
         workspace = get_object_or_404(
             Workspace,
             project__org__slug=self.kwargs["org_slug"],
             project__slug=self.kwargs["project_slug"],
             name=self.kwargs["workspace_slug"],
         )
 
-        # TODO: check permissions here
+        # only a privileged user can view the current files
+        if not has_permission(
```

Comment:
Definitely agree with this in principle but there are some stricky parts:
1) the exceptions raised are DRF specific and use API specific language
2) the anon check is repeated (for good reason in that function!) before the has_permission function

I'd like to add a few more of these calls, getting the Roles and Permissions we use for gates settled, and then see if we can extract a nice abstraction across all of them.  I think we're still working out some of those bits currently, but we should definitely make sure this is dealt with before the end of the voyage (captured in #699).

Reply:
Good point.

I think we should probably move into `jobserver/releases.py`, but raise custom logic Exceptions, that the views/DRF can handle as needed.

If `has_permission` checks for anon, then it can be removed for now.